wolfcrypt/src/misc.c

```diff
@@ -187,11 +187,31 @@ WC_STATIC WC_INLINE word32 ByteReverseWord32(word32 value)
 WC_STATIC WC_INLINE void ByteReverseWords(word32* out, const word32* in,
                                     word32 byteCount)
 {
-    word32 count = byteCount/(word32)sizeof(word32), i;
-
-    for (i = 0; i < count; i++)
-        out[i] = ByteReverseWord32(in[i]);
+    word32 count, i;
 
+#ifdef WOLFSSL_USE_ALIGN
+    if ((((unsigned long)in & 0x3) == 0) &&
+        (((unsigned long)out & 0x3) == 0))
+    {
+#endif
+        count = byteCount/(word32)sizeof(word32);
+        for (i = 0; i < count; i++)
+            out[i] = ByteReverseWord32(in[i]);
+#ifdef WOLFSSL_USE_ALIGN
+    } else {
+        byte *in_bytes = (byte *)in;
+        byte *out_bytes = (byte *)out;
+
+        byteCount &= ~0x3U;
+
+        for (i = 0; i < byteCount; i += sizeof(word32)) {
```

Comment:
Doesn't work when in == out.

Try doing word32 but load in[i] into volatile temporary.

Reply:
>Try doing word32 but load in[i] into volatile temporary.

Ha, I coded it that way first, then redid for efficiency, forgetting the args weren't `restrict`.  Good catch.

Reply:
Sorry I've been playing and found it didn't work.
I updated the comment.